lib/iris/_cube_coord_common.py

```diff
@@ -34,7 +34,7 @@ class LimitedAttributeDict(dict):
                        'coordinates', 'grid_mapping', 'climatology',
                        'cell_methods', 'formula_terms', 'compress',
                        'missing_value', 'add_offset', 'scale_factor',
-                       'valid_max', 'valid_min', 'valid_range', '_FillValue')
```

Comment:
> Sure. In practice this means checking and or coercing to ensure the following statement holds true:
> "The type of each valid_range, valid_min and valid_max attribute should match the type of its variable (except that for byte data, these can be of a signed integral type to specify the intended range)."

This is slightly odd, in my eyes, in that attribute values are not typed.
I don't know if there is an inferred typing convention regarding numbers, or where this is referenced.

Is the only difference here between integers and floats?

I'm not sure that explicitly altering an attribute value as part of the netCDF saving process is required here.
What do others think?